src/canvasController.ts

```diff
@@ -0,0 +1,91 @@
+import { CanvasValues } from './constants';
+import { Enemy } from './enemy';
+import { Player } from './player';
+import { Weapon } from './weapon';
+
+export class CanvasController {
+  private ctx: CanvasRenderingContext2D;
+  constructor(ctx: CanvasRenderingContext2D) {
+    this.ctx = ctx;
+  }
+
+  private drawPlayer(player: Player) {
+    this.ctx.save();
+    this.ctx.translate(player.body.x, player.body.y);
+    this.ctx.rotate(player.body.heading);
+
+    this.ctx.fillStyle = player.color;
+    this.ctx.fillRect(
+      (-1 * player.body.width) / 2,
+      (-1 * player.body.height) / 2,
+      player.body.width,
+      player.body.height
+    );
+
+    this.ctx.fillStyle = player.faceColor;
+    this.ctx.fillRect(
+      (-1 * player.body.width) / 2,
+      player.body.height / 2 - 2,
+      player.body.width,
+      2
+    );
+
+    if (player.isAttacking) {
+      const weaponRotation =
+        (player.weapon.swingAngle * (Date.now() - player.attackTimer)) /
+        player.weapon.attackDuration;
+      this.drawWeapon(player.weapon, weaponRotation, player.body.width / 2);
+    }
+
+    this.ctx.restore();
+  }
+
+  private drawEnemy(enemy: Enemy) {
+    this.ctx.fillStyle = 'red';
+    this.ctx.fillRect(
+      enemy.body.x - enemy.body.width / 2,
+      enemy.body.y - enemy.body.height / 2,
+      enemy.body.width,
+      enemy.body.height
+    );
+  }
+
+  private drawWeapon(
+    weapon: Weapon,
+    rotation: number,
+    distanceFromUser: number
+  ) {
+    this.ctx.rotate(weapon.swingAngle / 2 - rotation);
+    this.ctx.translate(0, weapon.width / 2);
+
+    this.ctx.fillStyle = 'black';
```

Comment:
weapon.color?

Reply:
I'm uncertain about this. Do you believe I should include the 'weapon.color' property for now, even though it might become obsolete once sprites are introduced?

Reply:
Nevermind this. Weapons now have colors lol 787e9cd6e700a0463045e5e409093fd21e9216f7